electrum/lnonion.py

```diff
@@ -157,6 +158,10 @@ def from_bytes(cls, b: bytes) -> 'OnionPacket':
             version=b[0],
         )
 
+    @property
+    def onion_hash(self) -> bytes:
+        return sha256(self.to_bytes())
```

Comment:
not sure how expensive this is, just noting that it could conceivably be `@cached_property`